### PR TITLE
k8s support diff mode

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

``diff`` attribute was always returned in output when ``k8s`` module was executed using options, ``force``, ``apply`` or ``state=patched`` even if the ``diff=no`` was set at task level.
This fix aims to correct this
#88 

This fix is not working if collection ``cloud.common`` is used without the fix https://github.com/ansible-collections/cloud.common/pull/65

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

k8s

### CODE REVIEW

File: changelogs/fragments/146-k8s-add-support-diff-mode.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - k8s - Add support for diff mode, diff is returned only when it is explicitly requested (https://github.com/ansible-collections/kubernetes.core/pull/146).
```
Comment (gravesm) [inline changelogs/fragments/146-k8s-add-support-diff-mode.yml:2]:
```suggestion
  - k8s - Fixes a bug where diff was always returned when using apply or modifying an existing object, even when diff=no was specified. The module no longer returns diff unless requested and will now honor diff=no (https://github.com/ansible-collections/kubernetes.core/pull/146).
```

**Review by gravesm — CHANGES_REQUESTED** [review]
Just realized we also need to fix the `k8s_scale` and `k8s_json_patch` modules, too.

**Comment by gravesm** [comment]
> Just realized we also need to fix the k8s_scale and k8s_json_patch modules, too.

This still needs to be addressed.

**Comment by abikouo** [comment]
> > Just realized we also need to fix the k8s_scale and k8s_json_patch modules, too.
> 
> This still needs to be addressed.

done

**Review by gravesm — APPROVED** [review]
Thanks!

**Pull request merged**